st3/sublime_lib/settings_listener.py

```diff
@@ -145,13 +145,29 @@ def on_setting_changed(
 
     The handler is not called every time the `settings` object changes,
     but only when the value derived using the `selector` argument changes.
-    If `selector` is callable, then the derived value is ``selector(self)``.
+    If `selector` is callable, then the derived value is ``selector(settings)``.
     If `selector` is a :class:`str`,
-    then the derived value is ``self.get(selector, None)``.
-    Otherwise, the derived value is ``projection(self, selector)``.
+    then the derived value is ``settings.get(selector, None)``.
+    Otherwise, the derived value is ``projection(settings, selector)`` (as defined below).
 
     The handler should accept two arguments:
     the new derived value and the previous derived value.
+
+    ``projection(settings, keys)``
```

Comment:
Since we use selectors twice in `sublime_lib`, I really believe this explanation should go into its own section that we then reference. Kind of how the official docs have a [types section](https://www.sublimetext.com/docs/api_reference.html#types) that they link to for such special types.

Because `projection` isn't a function directly used by the user of this API and they can't change the first parameter, I believe it makes more sense to explain the effects of the second parameter only. I just don't think think the technical explanation is useful.

Reply:
Sorry, my brain's been stuck in a loop on the `projection` docs. I agree there's room for improvement here, but is it good enough to release? If not, then it might be simplest to limit `@on_setting_changed` to a string or callable for now.

Reply:
Github doesn't show my comment to me anymore, so I'm adding another one just in case. 

It's probably good enough, but I added a separate issue to track this so we don't forget.

Also, kinda unrelated, but it would have been good if you removed the `[WIP]` tag in the PR title at some point. Since I've been struggling to find time for open source lately, I need to select carefully which issues and PRs I take a look into first.